### ISSUE TITLE
Rule proposal: `prefer-ternary`

### ISSUE THREAD

**Issue by jmoore914** [sindresorhus/eslint-plugin-unicorn#486]
This rule would enforce using ternary expressions to simple if/then/else statements.

# Fail
```js
if(condition){
  a = 1;
}
else{
  a = 2;
}
```
# Pass
```js
a = condition ? 1 : 2
```
```js
if(condition){
  doSomeStuff()
  a = 1;
}
else{
  a = 2;
}
```

This rule could have options for whether it would be enabled for various situations (e.g. variable assignments, return statements, function calls).

I've already written this rule as a normal ESLint plugin so it would just need some minor rewriting to conform to the unicorn linting/documentation

**Comment by sindresorhus** [comment]
Accepted

**Comment by jmoore914** [comment]
Progress so far: https://github.com/jmoore914/eslint-plugin-unicorn/tree/prefer-ternary

Just need to find time to add documentation which might take a while because of all the different options available.
